### PR TITLE
Feature: testing

### PR DESCRIPTION
**Pull request by meme-es**
## Testing
This pull request delivers test specs over the public methods of the project:

### Public methods of Player:
```
#name
#character
#score
```
### Public methods of Game
```
#move
#free?
#board
#ended?
#result
```

### CODE REVIEW

File: spec/tic_tac_toe_spec.rb [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+RSpec.describe Game do
+  let(:game) { Game.new }
+  let(:player) { Player.new('Manuel', 'X') }
+
+  describe '#move' do
+    context 'when the method is called' do
+      it 'returns nil always' do
+        expect(game.move(player, 5)).to be(nil)
+      end
+    end
```
Comment (maelfosso) [inline spec/tic_tac_toe_spec.rb]: Testing the `move` method in your situation will result to compare the board content, not the return value
```ruby
game.move(player, 5)
expect(game.board). to be([....])
```

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required :red_circle: 

Hello, 
You have written an amazing test, simply to read. Thank for that.
However, there are some changes to do

1. I run the test and I got this error
```ruby
$ rspec spec/
Please introduce the first player's name: F..............

Failures:

  1) UserInterface#play when the method is called returns nil allways
     Failure/Error: new_name = gets.chomp
     
     Errno::EISDIR:
       Is a directory @ io_fillbuf - fd:5 spec/
     # ./lib/userinterface.rb:42:in `gets'
     # ./lib/userinterface.rb:42:in `gets'
     # ./lib/userinterface.rb:42:in `block in choose_name'
     # ./lib/userinterface.rb:40:in `loop'
     # ./lib/userinterface.rb:40:in `choose_name'
     # ./lib/userinterface.rb:33:in `new_player'
     # ./lib/userinterface.rb:6:in `play'
     # ./spec/tic_tac_toe_spec.rb:13:in `block (4 levels) in <top (required)>'

Finished in 0.0074 seconds (files took 0.08168 seconds to load)
15 examples, 1 failure

Failed examples:

rspec ./spec/tic_tac_toe_spec.rb:12 # UserInterface#play when the method is called returns nil allways
```
You can write test for **I/O method**, that is for `UserInterface#play`.

2. Please, move the test of each class in their own file.

3. Read the comment below.

Kindly make those changes and submit a new code review request.
Thanks

Mael FOSSO

File: spec/player_spec.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+require './lib/player'
+
+RSpec.describe Player do
+  let(:player) { Player.new('Manuel', 'X') }
+
+  describe '#name' do
+    context 'when the methodd is called' do
+      it 'returns the string name' do
+        expect(player.name).to be_an_instance_of(String)
```
Comment (mosesogwo) [inline spec/player_spec.rb]: You should add another test to check if the string returned is 'Manuel'

File: spec/player_spec.rb [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      it "doesn't return another instance than a String" do
+        expect(player.name).not_to be_an_instance_of(Integer)
+      end
+    end
+  end
+
+  describe '#character' do
+    context 'when the methodd is called' do
+      it 'returns the symbol string of the player' do
+        expect(player.character).to be_an_instance_of(String)
```
Comment (mosesogwo) [inline spec/player_spec.rb]: You should add a test to check that the string returned is exactly the player's character

File: spec/game_spec.rb [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        game.move(player, 5)
+        expect(game.free?(5)).not_to be(true)
+      end
+    end
+  end
+
+  describe '#board' do
+    context 'when the method is called' do
+      it 'returns a string with the current board status' do
+        expect(game.board).to be_an_instance_of(String)
```
Comment (mosesogwo) [inline spec/game_spec.rb]: You should add test to check that the string returned correctly represents the state of the board.

**Review by mosesogwo — CHANGES_REQUESTED** [review]
**Hi @meme-es, @kbjude :clap:. Good job:+1:, but you haven't effected all the changes requested from the previous review**.
- In testing the **move** method, you should check the status of the board after the move has been made; not just the return value. See the previous reviewer's comment on this.
- In addition to the above, for your player_spec, you should check if the return value is correct and not just the class of the return value. See comments on your spec file below.

_**Kindly make these changes and submit another code review request.**_

**Any question? You can reach me on Slack (Moses Ogwo)**

**Cheers:stars:**

### Status: Changes Requested.

**Review by maelfosso — APPROVED** [review]
# Congratulations :clap: 

You can merge it :+1: 
Good luck with the next :handshake: 

Mael FOSSO

**Comment by meme-es** [comment]
Thank you @maelfosso & @mosesogwo for your help.

**Pull request merged**